grafana-plugin/src/containers/Rotation/Rotation.tsx

```diff
@@ -104,8 +129,11 @@ const Rotation: FC<RotationProps> = (props) => {
                     currentTimezone={currentTimezone}
                     color={color}
                     handleAddOverride={getAddOverrideClickHandler(event)}
+                    handleAddShiftSwap={getAddShiftSwapClickHandler(event)}
+                    onShiftSwapClick={onShiftSwapClick}
                     simplified={simplified}
                     filters={filters}
+                    onClick={onSlotClick && getSlotClickHandler(event)}
```

Comment:
Can we not check for the existance of `onSlotClick` inside the handler instead? Or is there a strong reason for this approach? Usually it is preferred to check against any guards within the function and not outside, as if that would be reused the developer using it could forget about the required checks.